traitsui/tests/test_editor.py

```diff
@@ -0,0 +1,733 @@
+from traitsui.editor import Editor
+from traitsui.editor_factory import EditorFactory
+from traitsui.handler import default_handler
+from traitsui.ui import UI
+
+from ._tools import is_current_backend_null
+
+
+class FakeControl(HasTraits):
+
+    #: The value stored in the control.
+    control_value = Any
+
+    #: An event which also can be fired.
+    control_event = Event
+
+
+class StubEditorFactory(EditorFactory):
+
+    #: Whether or not the traits are events.
+    is_event = Bool()
+
+    #: Whether or not the traits are events.
+    auxiliary_value = Any(sync_value=True)
+
+    #: Whether or not the traits are events.
+    auxiliary_cv_int = CVInt
+
+    #: Whether or not the traits are events.
+    auxiliary_cv_float = CVFloat
+
+    def _auxiliary_cv_int_default(self):
+        return 0
+
+    def _auxiliary_cv_float_default(self):
+        return 0.0
+
+
+class StubEditor(Editor):
+
+    #: Whether or not the traits are events.
+    is_event = Bool
+
+    #: An auxiliary value we want to synchronize.
+    auxiliary_value = Any
+
+    #: An auxiliary list we want to synchronize.
+    auxiliary_list = List
+
+    #: An auxiliary event we want to synchronize.
+    auxiliary_event = Event
+
+    #: An auxiliary int we want to synchronize with a context value.
+    auxiliary_cv_int = Int(sync_value='from')
+
+    #: An auxiliary float we want to synchronize with a context value.
+    auxiliary_cv_float = Float
+
+    def init(self, parent):
+        self.control = FakeControl()
+        self.is_event = self.factory.is_event
+        if self.is_event:
+            self.control.on_trait_change(self.update_object, 'control_event')
+        else:
+            self.control.on_trait_change(self.update_object, 'control_value')
+
+    def dispose(self):
+        if self.is_event:
+            self.control.on_trait_change(self.update_object, 'control_event',
+                                         remove=True)
+        else:
+            self.control.on_trait_change(self.update_object, 'control_value',
+                                         remove=True)
+        super(StubEditor, self).dispose()
+
+    def update_editor(self):
+        if self.is_event:
+            self.control.control_event = True
+        else:
+            self.control.control_value = self.value
+
+    def update_object(self, new):
+        if self.control is not None:
+            if not self._no_update:
+                self._no_update = True
+                try:
+                    self.value = new
+                finally:
+                    self._no_update = False
+
+    def set_focus(self, parent):
+        pass
+
+
+class UserObject(HasTraits):
+
+    #: The value being edited.
+    user_value = Any('test')
+
+    #: An auxiliary user value
+    user_auxiliary = Any(10)
+
+    #: An list user value
+    user_list = List(['one', 'two', 'three'])
+
+    #: An event user value
+    user_event = Event
+
+
+@unittest.skipIf(is_current_backend_null(),
```

Comment:
Might be worth having a "requires_non_null_backend" decorator in `_tools` for this sort of thing. (Naming to be bikeshedded, and not as part of this PR.)

Reply:
There is in fact such a decorator, but it only works with methods and not with classes at the moment :(